paddle/fluid/operators/recurrent_op.cc

```diff
@@ -636,10 +636,6 @@ class RecurrentGradOpShapeInference : public framework::InferShapeBase {
 
     // In some case the kInitialStates is empty.
     if (ctx->HasInputs(RecurrentBase::kInitialStates)) {
-      PADDLE_ENFORCE_EQ(ctx->HasOutputs(framework::GradVarName(
-                            RecurrentBase::kInitialStates)),
-                        true, "The output of(%s) should not be empty.",
-                        framework::GradVarName(RecurrentBase::kInitialStates));
       ctx->SetOutputsDim(framework::GradVarName(RecurrentBase::kInitialStates),
```

Comment:
Although I asked you to delete this PADDLE_ENFORCE_EQ line, you should also read some lines above and below. The lines below are Potential dangerous code.

If there is no GradVarName, you shouldn't SetOutputsDim for the grad of InitialStates. You should change the "if" condition to

if (ctx->HasInputs(RecurrentBase::kInitialStates) && ctx->HasOutputs(framework::GradVarName(RecurrentBase::kInitialStates))  {
    SetOutputsDim ...
}

Reply:
Done.